aten/src/ATen/ThreadLocalDebugInfo.h

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <c10/macros/Export.h>
+
+#include <memory>
+#include <string>
+
+namespace at {
+
+class CAFFE2_API ThreadLocalDebugInfoBase {
```

Comment:
Can't we just use shared_ptr<void> for it? If we wanted more safety, we could use our own RTTI (based on TypeMeta) but it'd be a bit weird use case for it.